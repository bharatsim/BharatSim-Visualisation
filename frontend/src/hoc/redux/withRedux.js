import React from 'react';
import { Provider } from 'react-redux';
import store from './store';

export default (WrappedComponent) => (props) =>
  (
    <Provider store={store}>
      <WrappedComponent {...props} />
    </Provider>
  );
