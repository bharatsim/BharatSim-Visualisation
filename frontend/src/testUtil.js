/* eslint-disable */
import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import reducer from './hoc/redux/reducer'
import { fireEvent, within } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { ProjectLayoutProvider } from './contexts/projectLayoutContext';

export function selectDropDownOption(container, dropDownId, optionId) {
  const dropDown = container.getByTestId(dropDownId);
  fireEvent.mouseDown(within(dropDown).getByRole('button'));
  const options = within(
    within(document.getElementById(`menu-${dropDownId}`)).getByRole('listbox'),
  );
  fireEvent.click(options.getByTestId(`${dropDownId}-${optionId}`));
}

export function withRouter(WrappedComponent) {
  const history = createMemoryHistory();
  return (props) => (
    <Router history={history}>
      <WrappedComponent {...props} />
    </Router>
  );
}

export function withProjectLayout(WrappedComponent) {
  return (props) => (
    <ProjectLayoutProvider
      value={{
        projectMetadata: {
          id: '1',
          name: 'project1',
        },
        selectedDashboardMetadata: {
          _id: '1',
          name: 'dashboard1',
        },
        addDashboard: jest.fn(),
        deleteDashboard: jest.fn(),
      }}
    >
      <WrappedComponent {...props} />
    </ProjectLayoutProvider>
  );
}



export function renderWithRedux(
    ui,
    {
        initialState={},
        store = createStore(reducer, initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}
