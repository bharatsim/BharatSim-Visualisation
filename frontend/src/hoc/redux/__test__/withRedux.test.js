import React from 'react';
import { useSelector } from 'react-redux';
import { renderWithRedux as render } from '../../../testUtil';
import withRedux from '../withRedux';

function DummyComponent() {
  const notifications = useSelector((state) => state.snackBar.notifications);
  return (
    <div id="testId">
      {'Dummy Component '}
      {notifications.length}
    </div>
  );
}

describe('withRedux', () => {
  const Component = withRedux(DummyComponent);
  it('should be able to access redux state', () => {
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
});
