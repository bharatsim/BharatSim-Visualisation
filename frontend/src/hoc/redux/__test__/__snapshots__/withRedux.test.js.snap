// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRedux should be able to access redux state 1`] = `
<div>
  <div
    id="testId"
  >
    Dummy Component 
    0
  </div>
</div>
`;
